### DISCUSSION TITLE
Unable to install Kimai2 on Debian 11 with MySQL 8

### DISCUSSION THREAD

**Discussion by blessleness** [kimai/kimai#3023] [On-Premise: Installation & Updates] [answered]
Dear community

Now that I had previously successfully installed KImai2 on a VM running on my laptop (Ubuntu with MariaDB), I want to put it into production on an already existing server I have that runs Moodle. The server runs Debian 11 Buster and uses MySQL 8 DB Server.

Folder is created in /var/www/html/kimai2
DB is created in MySQL with user "kimai2" and necessary privileges
Now I want to run the installation running the command
`bin/console/ kimai:install -n`

And I get following error message
` [ERROR] Failed to create database: An exception occurred in driver: SQLSTATE[HY000] [2054] The server requested
         authentication method unknown to the client
`

After some research, I found out that MySQL did change its default authentication method, but honestly I did not really understand how to fix this...

As mentionned, I cannot touch too much to MySQL Server since Moodle runs already, and I definitelly don't wanna harm this installation.

THanks a lot for your support

**Comment by delvalt** [comment, marked as answer]
Hi blessleness,

I too have encountered this problem, and here is how to solve it.

Login as root to your DB and run the following query:

`ALTER USER 'kimai2'@'{host}' IDENTIFIED WITH mysql_native_password BY '{password}';`

(make sure to replace `{host}` with the current host for kimai2 user, most likely localhost, and `{password}` with the current kimai2 user's password)

This will change the authentication method for kimai2 user only (thus not impacting anything else on the server) to use the native password auth method instead of auth_socket (which is now the default since MySQL 8.0).

**Reply by kevinpapst** [reply]
Wow, that is a hard BC break that MySQL introduced there. 
Thanks for sharing the solution 👍 if this happens more often, I will update the docs.

**Reply by delvalt** [reply]
You're welcome! Yes, updating the doc seems like a good idea, it's likely that more and more people will install/upgrade MySQL to 8.0 version or higher.

**Reply by blessleness** [reply]
I confirm, the tip is working, Thx delvalt !
